### ISSUE TITLE
Ray Job Plugin 

### ISSUE THREAD

**Issue by shuangkun** [argoproj/argo-workflows#13838]
# Summary

Ray Job Plugin to manage Ray Jobs.

## Use Cases

Ray has become a new generation framework for data processing in large language model scenarios. Can we add corresponding plugins to better submit and manage Ray Jobs?

---

**Message from the maintainers**:

Love this feature request? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by Joibel** [comment]
This repository doesn't implement plugins directly, I'd suggest implementing this outside of this repository and linking the plugin to the documentation once done.

**Comment by shuangkun** [comment]
Yes, this repository will not implement this, but I haven't thought of where I can organize relevant people who are interested in this issue to do this. So I raised an issue to put it here.

**Comment by Joibel** [comment]
> Yes, this repository will not implement this, but I haven't thought of where I can organize relevant people who are interested in this issue to do this. So I raised an issue to put it here.

Of course, sounds good.

**Comment by shuangkun** [comment]
https://github.com/shuangkun/argo-workflows-ray-plugin  An example of a Ray plugin. We will continue to improve it in the future, adding Ray Cluster, Ray Service, etc.
If anyone is interested, welcome to co-build it. We can improve it and contribute it to argoproj-labs.

**Comment by Joibel** [comment]
If you'd like a repo in argoproj-labs I am happy to set this up for you.

**Comment by shuangkun** [comment]
Thanks a lot, it would be great if it could be under argoproj-labs

**Comment by Joibel** [comment]
I've created https://github.com/argoproj-labs/argo-workflows-ray-plugin for you, and you're an admin on that repo @shuangkun

**Comment by shuangkun** [comment]
> I've created https://github.com/argoproj-labs/argo-workflows-ray-plugin for you, and you're an admin on that repo [@shuangkun](https://github.com/shuangkun)

Great！ Thanks

**Comment by zylim-ml** [comment]
Hi, may I know if the latest version of [https://github.com/argoproj-labs/argo-workflows-ray-plugin](https://github.com/argoproj-labs/argo-workflows-ray-plugin) able to spin up a ray cluster and run ray train/ tune job?

**Comment by shuangkun** [comment]
> Hi, may I know if the latest version of [https://github.com/argoproj-labs/argo-workflows-ray-plugin](https://github.com/argoproj-labs/argo-workflows-ray-plugin) able to spin up a ray cluster and run ray train/ tune job? 

 Oh Not yet,Only RayJob Now. would you interest in to implement? It is not complicated

**Comment by Joibel** [comment]
Closing as this is now in labs as commented above.
